api_app/api/routes/resource_helpers.py

```diff
@@ -21,9 +23,13 @@ async def save_and_deploy_resource(resource: Resource, resource_repo: ResourceRe
     try:
         resource.user = user
         resource.updatedWhen = get_timestamp()
-        resource_repo.save_item(resource)
+
+        # Making a copy to save with secrets masked
+        masked_resource = deepcopy(resource)
+        masked_resource.properties = mask_sensitive_properties(resource.properties, resource_template)
+        resource_repo.save_item(masked_resource)
```

Comment:
Note: I'm doing the masking in this method and not in `save_item` because here is where I have the template available. `save_item` does not currently accept template as an argument, and I thought changing it isn't worth it. Open to suggestions, though.